spectrum-offchain/src/backlog.rs

```diff
@@ -360,7 +360,9 @@ where
     where
         TOrd::TOrderId: 'a,
     {
-        self.store.exists(ord_id).await
+        self.pending_pq.iter().any(|w| w.0.order_id == ord_id)
```

Comment:
This is a bug fix: when checking existence of an order we should check that they belong in the queues, not the persistent storage.

Reply:
Why? If an order exists in storage it must exist in one of the queues.

Reply:
It It doesn't happen in practice. For example I'd the not accepts the tx for an order, we don't call' remove' on it. 

We need to tighten this up.

Reply:
It seems to me that you're trying to cover the bug with more code instead of fixing the root bug that breaks the above assumption.

Reply:
Reverted this change.